camelot/view/proxy/collection_proxy.py

```diff
@@ -510,16 +511,20 @@ def model_run(self, model_context):
                 columns = tuple(moves.xrange(len(model_context.static_field_attributes)))
                 self.changed_ranges.extend(self.add_data(model_context, row, columns, obj, True))
                 updated_objects.add(subsystem_obj)
-                updated_objects.update(set(admin.get_depending_objects(obj)))
+                depending_objects = set(admin.get_depending_objects(obj))
+                updated_objects.update(depending_objects)
+                deleted_objects.update(depending_objects_before_set-depending_objects)
```

Comment:
hmm, die delta zijn niet noodzakelijk deleted objects, dat kunnen ook object zijn waarnaar de relatie gewoon is weggevallen.

zou het niet voldoende zijn om in de updated_objects de union te steken van de depending_objects before and after the change ?  een gevolg van een update kan zijn dat een object nt meer in een bepaalde collectie voorkomt.

Reply:
Dat is niet voldoende denk ik. In de case waarbij ge dan een role verwijdert wordt dat niet gepropageerd naar de proxy, waardoor dat object enkel in de _object weg is, maar de indexering nog bestaat. Bij het daarna aanmaken van een nieuw object wordt dat toegevoegd op de index van het oude object, waarbij de indexering nog overeenkomt met het oude object, met als gevolgd dat het nieuwe object niet meteen zichtbaar is in de UI.

Reply:
Een mogelijkheid is om via de related admin van de depending objects union te kijken of het object verwijderd is via de is_deleted methode, en dan afhankelijk daarvan het te updaten/deleten?

Reply:
> Een mogelijkheid is om via de related admin van de depending objects union te kijken of het object verwijderd is via de is_deleted methode, en dan afhankelijk daarvan het te updaten/deleten?

Dat lijkt op basis van de eerste testen alvast te werken...

Reply:
Zie 3bcb44b